### ISSUE TITLE
Connecting with new agent returns "The URI prefix is not recognized."

### ISSUE THREAD

**Issue by Ne0nd0g** [s3inlc/hashtopussy#280]
#### Before you submit an issue please include the following information if you do not your issue will be closed.

**Your current Server version located at the bottom of any hashtopussy webpage.**
`0.4.2`
**Current Client version**
`0.45.8`
**Your current Hashcat version**
`3.6.0`
**The exact task command you are trying to run.**
 `mono hashtopussyclient.exe`
**Debug output from the client by running "hashtopussy.exe debug"**
```
tom@hashcracker002:/opt/hashtopussyclient$ mono hashtopussyclient.exe debug
Client Version 0.45.8
Please enter server connect URL (https will be used unless specified):
http://172.16.5.60/api/server.php
Testing connection to http://172.16.5.60/api/server.php
{"action":"test"}

Unhandled Exception:
System.NotSupportedException: The URI prefix is not recognized.
  at System.Net.WebRequest.Create (System.Uri requestUri, System.Boolean useUriBase) [0x000ae] in <d2c057d9d34d4e029e580897bd60340c>:0
  at System.Net.WebRequest.Create (System.String requestUriString) [0x00014] in <d2c057d9d34d4e029e580897bd60340c>:0
  at jsonClass.jsonSendOnce (System.String json) [0x00006] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.initConnect () [0x00065] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.Main (System.String[] args) [0x0008b] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.NotSupportedException: The URI prefix is not recognized.
  at System.Net.WebRequest.Create (System.Uri requestUri, System.Boolean useUriBase) [0x000ae] in <d2c057d9d34d4e029e580897bd60340c>:0
  at System.Net.WebRequest.Create (System.String requestUriString) [0x00014] in <d2c057d9d34d4e029e580897bd60340c>:0
  at jsonClass.jsonSendOnce (System.String json) [0x00006] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.initConnect () [0x00065] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.Main (System.String[] args) [0x0008b] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
tom@hashcracker002:/opt/hashtopussyclient$
```
**Describe your problem in as much detail as possible " It's broke " is not a description.**
Trying to register a new agent. The server is active an accepts connections. 
```
tom@hashcracker002:/opt/hashtopussyclient$ mono hashtopussyclient.exe
Client Version 0.45.8
Please enter server connect URL (https will be used unless specified):
http://172.16.5.60/api/server.php
Testing connection to http://172.16.5.60/api/server.php

Unhandled Exception:
System.NotSupportedException: The URI prefix is not recognized.
  at System.Net.WebRequest.Create (System.Uri requestUri, System.Boolean useUriBase) [0x000ae] in <d2c057d9d34d4e029e580897bd60340c>:0
  at System.Net.WebRequest.Create (System.String requestUriString) [0x00014] in <d2c057d9d34d4e029e580897bd60340c>:0
  at jsonClass.jsonSendOnce (System.String json) [0x00006] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.initConnect () [0x00065] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.Main (System.String[] args) [0x0008b] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.NotSupportedException: The URI prefix is not recognized.
  at System.Net.WebRequest.Create (System.Uri requestUri, System.Boolean useUriBase) [0x000ae] in <d2c057d9d34d4e029e580897bd60340c>:0
  at System.Net.WebRequest.Create (System.String requestUriString) [0x00014] in <d2c057d9d34d4e029e580897bd60340c>:0
  at jsonClass.jsonSendOnce (System.String json) [0x00006] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.initConnect () [0x00065] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
  at hashtopussy.Program.Main (System.String[] args) [0x0008b] in <58c1b76d6bdb4ca3937046d822f9eae1>:0
tom@hashcracker002:/opt/hashtopussyclient$
```
Web Server responds just fine:
```
tom@hashcracker002:/opt/hashtopussyclient$ curl http://172.16.5.60/api/server.php
{"action":"INV","response":"ERROR","message":"Invalid query!"}
tom@hashcracker002:/opt/hashtopussyclient$
```

**Comment by Ne0nd0g** [comment]
I have a different agent that is working fine and had no problems registering. I manually created the `URL` file and put the server URL in it. This particular agents keeps going through this cycle over and over where it will not connect. `test001` is a valid voucher. I did notice this message `System.Collections.ListDictionaryInternal` wasn't present on the other agent that was working fine. I did a tcpdump and the application never even attempted to connect to the server.

Debug output from the client by running "hashtopussy.exe debug"
```
tom@hashcracker002:/opt/hashtopussyclient$ mono ./hashtopussy.exe debug
Client Version 0.45.8
Checking for client updates
{"action":"update","type":"csharp","version":"0.45.8"}
Could not connect to specified server, exiting
Attempting to re-connect in 2 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 3 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 4 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 7 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 8 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 8 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 11 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 14 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 11 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 11 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 18 seconds

System.Collections.ListDictionaryInternal
Empty string for success check
Unable to find existing token, please enter voucher
test001
System is Linux
{"action":"register","voucher":"test001","name":"hashcracker002","uid":"40832758-baff-4993-b10d-0Core Processor Family Integrated Graphics Controller (rev 09)"]}
Could not connect to specified server, exiting
Attempting to re-connect in 2 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 3 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 5 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 5 seconds
Could not connect to specified server, exiting
Attempting to re-connect in 8 seconds
```

**Comment by Ne0nd0g** [comment]
I was using Ubuntu 16.04.3. I removed Mono and installed it according to the mono application directions found at: http://www.mono-project.com/download/#download-lin. Everything worked fine after that.

**Comment by winxp5421** [comment]
Strange. May I ask did you follow the client install walkthrough on the hashtopussy wiki or was this an existing environment? 

Asking because maybe the wiki needs updating.

**Comment by Ne0nd0g** [comment]
It was an existing environment and I followed the Wiki again to make sure after I purged all the packaged. I think it comes down the version of mono being installed. The Wiki doesn't use t he same repos that are documents on mono's website install instructions. I guessing it came down to a code library problem between the versions of ubuntu and mono and that is what caused this error `System.Collections.ListDictionaryInternal`.
